dataRetriever/dataPool/proofsCache/proofsPool.go

```diff
@@ -0,0 +1,106 @@
+package proofscache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/pkg/errors"
+)
+
+// ErrMissingProof signals that the proof is missing
+var ErrMissingProof = errors.New("missing proof")
+
+// ErrNilProof signals that a nil proof has been provided
+var ErrNilProof = errors.New("nil proof provided")
+
+type proofsPool struct {
+	mutCache sync.RWMutex
+	cache    map[uint32]*proofsCache
+}
+
+// NewProofsPool creates a new proofs pool component
+func NewProofsPool() *proofsPool {
+	return &proofsPool{
+		cache: make(map[uint32]*proofsCache),
```

Comment:
why do we need to save them by shard?

a node in shard X should have proofs only for X and Meta

Reply:
i've added the option to save by shardID in a more general way; indeed, currently we use the proofs only for current shard, i haven't covered the cross shard or meta yet; will analyse this for a next PR